src/app/components/quick-connect-dialog/quick-connect-dialog.component.ts

```diff
@@ -0,0 +1,183 @@
+import { HttpClient } from '@angular/common/http';
+import { Component, Inject, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { Animations } from 'app/animations/animations';
+import { ConnectionChoice, SuggestedConnection } from 'app/models/connection';
+import { UserToken } from 'app/models/usertoken';
+import { ChannelPollingService } from 'app/services/channel-polling.service';
+import { RaidenService } from 'app/services/raiden.service';
+import { TokenPollingService } from 'app/services/token-polling.service';
+import { combineLatest, Observable, Subject, zip } from 'rxjs';
+import {
+    catchError,
+    filter,
+    finalize,
+    first,
+    map,
+    switchMap,
+    takeUntil,
+    tap,
+} from 'rxjs/operators';
+import { TokenInputComponent } from '../token-input/token-input.component';
+
+export interface QuickConnectDialogPayload {
+    readonly token: UserToken;
+}
+
+export interface QuickConnectDialogResult {
+    readonly token: UserToken;
+    readonly connectionChoices: ConnectionChoice[];
+}
+
+@Component({
+    selector: 'app-quick-connect-dialog',
+    templateUrl: './quick-connect-dialog.component.html',
+    styleUrls: ['./quick-connect-dialog.component.scss'],
+    animations: Animations.stretchInOut,
+})
+export class QuickConnectDialogComponent implements OnInit, OnDestroy {
+    @ViewChild(TokenInputComponent, { static: true })
+    private tokenInput: TokenInputComponent;
+
+    form: FormGroup;
+    initiatedWithoutToken = false;
+    suggestions: SuggestedConnection[] = [];
+    loading = false;
+    pfsError = false;
+
+    private ngUnsubscribe = new Subject();
+
+    constructor(
+        @Inject(MAT_DIALOG_DATA) data: QuickConnectDialogPayload,
+        private dialogRef: MatDialogRef<QuickConnectDialogComponent>,
+        private fb: FormBuilder,
+        private tokenPollingService: TokenPollingService,
+        private raidenService: RaidenService,
+        private http: HttpClient,
+        private channelPollingService: ChannelPollingService
+    ) {
+        this.initiatedWithoutToken = !data.token;
+        this.form = this.fb.group({
+            token: [data.token, Validators.required],
+            totalAmount: ['', Validators.required],
+        });
+    }
+
+    ngOnInit(): void {
+        this.subscribeToSuggestions();
```

Comment:
Nitpick: private methods with internal, explicit subscriptions aren't usually the recommended way to do that in Angular. Maybe you could just initialize private cold observables, possibly `share`d, and then use `| async` on template, which does handle subscriptions automatically. Declarative is usually better than imperative.

Reply:
Makes totally sense! I like it your way better. Thanks!

Reply:
Couldn't make it because there were too many changes needed in the tests. Not worth the effort.